### PR TITLE
Upgraded to Table Schema spec-v1 null/types/constraints semantics

### LINKED ISSUE #50: Sync with specs-v1
# Overview

There is a many new stuff in http://specs.frictionlessdata.io/table-schema/ (v1) that should be analyzed and implemented.

# Tasks

- [x] port python implementation
- [x] ensure `geojson` type is supported (#24)

### PR DESCRIPTION
**Pull request by roll**
- fixes #50

---

That's just a straight port of https://github.com/frictionlessdata/tableschema-py/pull/159

This changes include:
- rebase on spec-v1 null/types/constraints semantics
- porting comprehensive python types/constraints testsuite (and it almost 100% passes)
- fixing many errors in implementation (old types was just incorrect in many aspects - accepting other types as a input value instead of only field type/string, cast floats as integers, cast numbers to strings (?), cast date/time with wrong ISO8601 format etc)
- adding all spec-v1 types which was missing (year, duration etc)
- passing all relevant tests that the codebase has before this PR

I would say there was too less tests for `Table/Schema/Field`. I was not working on it in this PR. So I'll address it in #53

PR commits are ready to be reviewed one by one.

### CODE REVIEW

**Comment by roll** [comment]
@amercader 
please take a look

File: src/profiles/geojson.json [hunk 1]
```diff
@@ -0,0 +1,208 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "id": "https://raw.githubusercontent.com/fge/sample-json-schemas/master/geojson/geojson.json#",
```
Comment (amercader) [inline src/profiles/geojson.json]: Not sure if you investigated them thoroughly but this one seems to be a bit more compliant:

https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/geojson.json

eg it allows position to have more than 2 values (GeoJSON allows a third one for elevation). Both allow the `crs` object which has been removed in the latest GeoJSON spec version, but that should be all right.

Comment (roll) [inline src/profiles/geojson.json]: Thanks. Will do in #53.

**Comment by roll** [comment]
@amercader 
Thanks!

About `yearmonth` - here is already fixed version per https://pre-v1.frictionlessdata.io/json-table-schema/#gyearmonth and our decision to use tuple to represent it. Or it's other issue?

**Pull request merged**